### PR TITLE
Json walkthrough

### PR DESCRIPTION
**Pull request by CubeOfCheese**
Screen Shot 2020-02-29 at 10 23 56 PM
Screen Shot 2020-02-29 at 10 24 16 PM
Picture shows the bio section was moved up to expose referrals

### CODE REVIEW

**Comment by IngridFeng** [comment]
Heyy Billy can you resolve the conflict in the two files first? Usually you want to choose one of the two blocks and delete the other. In the rare case, you might want to pick the contents that you think should stay in BOTH blocks.

**Comment by IngridFeng** [comment]
Nevermind - it's resolved. Great job!

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -27,6 +27,17 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 function getMessage() {
-    fetch('/data').then((response) => response.text())
-    .then((message) => document.getElementById('fetch-message').innerText = message);
+    fetch('/data').then((response) => response.json())
+    .then((message) => {
+        for (var i=0; i<message.length; i++) {
```
Comment (IngridFeng) [inline portfolio/src/main/webapp/script.js]: nit: we want space in between. 
for (var i = 0; i < message.length; i++) {

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,10 +25,20 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    ArrayList<String> referrals = new ArrayList<String>();
```
Comment (IngridFeng) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: indentation minus two spaces in front of the line.

File: portfolio/pom.xml [hunk 3]
```diff
@@ -23,6 +23,11 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+        <groupId>com.google.code.gson</groupId>
```
Comment (IngridFeng) [inline portfolio/pom.xml]: Nice job of introducing new libraries into your project!!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -24,10 +25,20 @@
...
+
+  public DataServlet() {
+    referrals.add("{\"author\": \"Michael\", \"content\": \"An example referral here\"}");
+    referrals.add("{\"author\": \"Jenny\", \"content\": \"An example referral here\"}");
+    referrals.add("{\"author\": \"Hana\", \"content\": \"An example referral here\"}");
+  }
+
```
Comment (IngridFeng) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I might be wrong - I think overriding init() is usually preferred than using an constructor, according to https://javapapers.com/servlet/why-not-declare-a-constructor-in-servlet/. But Brian should have more context about this.

Comment (IngridFeng) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Another way you might want to get this data organized - just a suggestion, you don't have to change it. And if you do, change it next week - enough fantastic work for this week.
https://github.com/google/codeu-starter-project/tree/master/src/main/java/com/google/codeu
If you look at the relationship between data and servlet directory here, notice how data contains the structure of our info like message. So you might want to introduce a new file called "referral.java" and class Referral can contains two fields - author, content. And then you need to add "get" and "set" method for each.
And then you can simply put "referrals.add(new Referral("Michael", "An example referral here"))"
General idea is we want to get our data organized such that when other people want to contribute to your code, it's clear to them where they should go.

Comment (CubeOfCheese) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: No the more comments the better. Thank you

**Pull request merged**